### ISSUE TITLE
Use box/spout instead of phpoffice/phpexcel

### ISSUE THREAD

**Issue by garak** [portphp/portphp#61]

**Comment by ddeboer** [comment]
Why? What are the advantages?

**Comment by garak** [comment]
Memory consumption on large data

**Comment by ddeboer** [comment]
Good point, @garak! Is this something you want to work on? We should create a separate portphp/spout repository to hold the code.

**Comment by garak** [comment]
OK

**Comment by ddeboer** [comment]
Created the repo at https://github.com/portphp/spout. @garak Please proceed by creating a PR against that repo. Thanks you very much!

**Comment by aaa2000** [comment]
@garak Did you start an implementation ?

**Comment by garak** [comment]
not yet

**Comment by infomaniac50** [comment]
I noticed the array buffering in portphp/excel. The next thing I was going to work was that. If PHPOffice/Spreadsheet has some kind of iterable interface, then we could use that instead of buffering with toArray()

**Comment by infomaniac50** [comment]
Of course PHPOffice has something like this. https://phpspreadsheet.readthedocs.io/en/develop/topics/accessing-cells/#looping-through-cells

**Comment by slashrsm** [comment]
Closing as obsolete for the current ecosystem.

The box/spout idea led to a separate portphp/spout repo, but that adapter was never finished. Meanwhile:

- `portphp/excel` (PHPExcel) is **deprecated/abandoned** in favor of **`portphp/spreadsheet`**
- `portphp/spreadsheet` **2.0.0** is the supported spreadsheet package (PhpSpreadsheet)

Memory/streaming improvements belong on spreadsheet (see also #89), not on reviving Spout/PHPExcel.

Feel free to open a new issue or PR on portphp/spreadsheet if you want streaming/chunked reads there.
